geniza/corpus/models.py

```diff
@@ -533,6 +536,108 @@ def index_data(self):
         # script+language when/if included in index data
     }
 
+    def merge_with(self, merge_docs, rationale, user=None):
+        """Merge the specified documents into this one. Combines all
+        metadata into this document, adds the merged documents into
+        list of old PGP IDs, and creates a log entry documenting
+        the merge, including the rationale."""
+
+        # initialize old pgpid list if previously unset
+        if self.old_pgpids is None:
+            self.old_pgpids = []
+
+        # if user is not specified, log entry will be associated with
+        # script and document will be flagged for review
+        script = False
+        if user is None:
+            user = User.objects.get(username=settings.SCRIPT_USERNAME)
+            script = True
+
+        description_chunks = [self.description]
+        language_notes = [self.language_note] if self.language_note else []
+        notes = [self.notes] if self.notes else []
+        needs_review = [self.needs_review] if self.needs_review else []
+
+        for doc in merge_docs:
+            # add merge id to old pgpid list
+            self.old_pgpids.append(doc.id)
+            # add any tags from merge document tags to primary doc
+            self.tags.add(*doc.tags.names())
+            # add description if set and not duplicated
+            if doc.description and doc.description not in self.description:
+                description_chunks.append(
+                    "Description from %s:\n%s" % (doc.shelfmark, doc.description)
+                )
```

Comment:
i realized on looking at the issue that adding the shelfmark indicator might not actually help...if we're merging two docs and they both have essentially the same shelfmark (because it's a join), then the description will say:
```
[primary description]
Description from ENA XXX + CUL YYY
[secondary description]
```
...which doesn't make much sense because _both_ docs are ENA XXX + CUL YYY. their shelfmark will be same. so we might want to change it to:
```
[primary description]
Description from PGPID [merged PGPID]
[secondary description]
```
...or something else. lmk if this doesn't make sense

Reply:
yeah, good point — I had it as pgpid at first and then saw in the issue we wanted shelfmark; but in all of these cases the shelfmark won't be helpful since they're all joins. I'll change it back to what I had before.

Reply:
were we still planning to change this back to PGPID rather than shelfmark?

Reply:
good catch! thank you, I'd forgotten about this one